### PR TITLE
Add readme command

### PR DESCRIPTION
**Pull request by kevmoo**
Share some code with the list command (and tests)

### CODE REVIEW

**Comment by kevmoo** [comment]
Pretty!

Package source                                        | Description                                                                                         | Published Version                                                                               
----------------------------------------------------- | --------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------
[mono_repo](mono_repo/)                               | CLI tools to make it easier to manage a single source repository containing multiple Dart packages. | [pub package](https://pub.dev/packages/mono_repo)
[test_flutter_pkg](test_flutter_pkg/)                 | A new Flutter project.                                                                              |                                                                                                 
[test_flutter_pkg_example](test_flutter_pkg/example/) |                                                                                                     |                                                                                                 
[test_pkg](test_pkg/)                                 |                                                                                                     |

File: mono_repo/lib/src/commands/list_command.dart [hunk 1]
```diff
@@ -96,13 +96,21 @@ Iterable<String> listPackages(
...
+Iterable<PackageConfig> enumeratePackages(
+  RootConfig rootConfig, {
+  required bool onlyPublished,
+}) =>
+    rootConfig.where((element) => !onlyPublished || element.pubspec._published);
+
+extension PubspecExtension on Pubspec {
+  bool get _published => version != null && publishTo != 'none';
+
+  String get pubBadge => _published
```
Comment (natebosch) [inline mono_repo/lib/src/commands/list_command.dart:113]: [optional] Consider moving to the file where it used.

File: mono_repo/lib/src/commands/readme_command.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    ['Package source', 'Description', 'Published Version'],
+  ];
+
+  for (var pkg in enumeratePackages(rootConfig, onlyPublished: onlyPublished)) {
+    rows.add([
+      '[${pkg.pubspec.name}](${pkg.relativePath}/)',
+      pkg.pubspec.description ?? '',
+      pkg.pubspec.pubBadge,
+    ]);
+  }
```
Comment (natebosch) [inline mono_repo/lib/src/commands/readme_command.dart:57]:
```suggestion
  final rows = [
    ['Package source', 'Description', 'Published Version'],
    for (var pkg in enumeratePackages(rootConfig, onlyPublished: onlyPublished))
      [
        '[${pkg.pubspec.name}](${pkg.relativePath}/)',
        pkg.pubspec.description ?? '',
        pkg.pubspec.pubBadge,
      ]
  ];
```

File: mono_repo/lib/src/commands/readme_command.dart [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    for (var row in rows) {
+      for (var i = 0; i < 3; i++) {
+        if (row[i].length > widths[i]) {
+          widths[i] = row[i].length;
+        }
+      }
+    }
+  }
+
+  for (var row in rows) {
```
Comment (natebosch) [inline mono_repo/lib/src/commands/readme_command.dart:70]: should this be inside the conditional? It's not likely to hurt to pad to `3` no matter what, but is it necessary?

**Review by natebosch — APPROVED** [review]

File: mono_repo/lib/src/commands/readme_command.dart (hunk 2 above)
Comment (kevmoo) [reply mono_repo/lib/src/commands/readme_command.dart:57]: Love it!

File: mono_repo/lib/src/commands/readme_command.dart (hunk 3 above)
Comment (kevmoo) [reply mono_repo/lib/src/commands/readme_command.dart:70]: ack

File: mono_repo/lib/src/commands/list_command.dart (hunk 1 above)
Comment (kevmoo) [reply mono_repo/lib/src/commands/list_command.dart:113]: But then I'd need to create another extension...or just a static thing. 🤷 

I hear ya...

Comment (natebosch) [reply mono_repo/lib/src/commands/list_command.dart:113]: Are you worried about the 2 lines of overhead? They could both be private unnamed extensions. You could come close to saving characters...

**Review by natebosch — APPROVED** [review]

File: mono_repo/lib/src/commands/list_command.dart (hunk 1 above)
Comment (kevmoo) [reply mono_repo/lib/src/commands/list_command.dart:113]: Don't smile at him mocking me, @jakemac53 !!

Comment (kevmoo) [reply mono_repo/lib/src/commands/list_command.dart:113]: okay. Fixing

**Pull request merged**